cmd/src/search_stream.go

```diff
@@ -21,20 +22,111 @@ func init() {
 }
 
 func streamSearch(query string, opts streaming.Opts, client api.Client, w io.Writer) error {
-	t, err := parseTemplate(streamingTemplate)
-	if err != nil {
-		panic(err)
+	var d streaming.Decoder
+	if opts.Json {
+		d = jsonDecoder(w)
+	} else {
+		t, err := parseTemplate(streamingTemplate)
+		if err != nil {
+			return err
+		}
+		d = textDecoder(query, t, w)
+	}
+	return streaming.Search(query, opts, client, d)
+}
+
+// jsonDecoder streams results as JSON to w.
+func jsonDecoder(w io.Writer) streaming.Decoder {
+	// check is used whenever we write to w. If a write fails, the output is most
+	// likely not a valid JSON and we should stop the stream immediately. Since the
+	// handlers of the decoder don't handle errors we use panic.
+	check := func(err error) {
+		if err != nil {
+			panic(err)
+		}
+	}
+	_, err := w.Write([]byte("{\"Results\":[\n"))
```

Comment:
why can't we just do line terminated json? ie we don't have a single json item, we have a json item per line. Then it is possible to pipe the output to jq. ie its more like a json encoded log file.

Reply:
Sure we could do that, but this would offload parsing work to the user because they have to distinguish between match events and metadata events like progress and alerts. If we stream out an object it is easier to work with.

Reply:
EDIT: I guess we just need to decide what the output format is. 1 big JSON object or a log-like file containing possibly thousands of objects, 1 per line.

Reply:
EDIT: Discussed offline with @keegancsmith, we go with log-like output, 1 result JSON-object per line.